### PR TITLE
(SIMP-10434) Add dockerfile to rebuild puppetserver on EL8

### PR DESCRIPTION
**Pull request by trevor-vaughan**
SIMP-10434 #close

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
This will output something like:

```
Your file is at:rpmbuild/RPMS/noarch/puppetserver-6.16.1-1.SIMP1.el8.noarch.rpm
```

File: puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# If you want to save your container for future use, you use use the `docker
+# commit` command
+#   * docker commit <running container ID> puppet_agent_build
```
Comment (lnemsick-simp) [inline puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile]: shouldn't this be called puppetserver_build instead?

File: puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#   * docker run -it puppet_agent_build
+#
+# If using buildah, you probably want to build this as follows since various
+# things might fail over time:
+#   * buildah bud --layers=true -f <filename> .
+#
+# You can choose to build a specific version of puppet by setting the
+# PUPPET_VERSION environment variable to a valid git reference in the
```
Comment (lnemsick-simp) [inline puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile]: I don't see how PUPPET_VERSION is used here

File: puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+RUN yum -y install https://yum.puppet.com/puppet6-release-el-8.noarch.rpm
+RUN yum -y install puppet-agent
+RUN yumdownloader puppetserver
+
+RUN rpm -i --nodigest --nofiledigest puppetserver*.rpm
+
+RUN dist=`rpm --eval '%{dist}'`; release=`rpm -q --qf="%{release}" puppetserver | cut -f1 -d'.'`; rpmrebuild --batch --release="${release}.SIMP1${dist}" puppetserver
+
+RUN echo '############'; echo -n "Your file is at:"; find rpmbuild/RPMS -name "*.rpm"; echo '##########'
```
Comment (lnemsick-simp) [inline puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile:33]: as a docker newbie, don't now how this helps me find the built RPM and then offload it from the container

File: puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile (hunk 1 above)
Comment (lnemsick-simp) [reply puppet_agent_build/Puppetserver_EL8_Rebuild.dockerfile]: If running a `docker build`on this is supposed to result in a running container, it doesn't, when I build this Dockerfile on EL7.  It results in a image, but not a running container that could be named with a `docker commit` command.

**Review by jeannegreulich — APPROVED** [review]

**Pull request merged**